include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -604,10 +606,8 @@ class __reduce_future
     auto
     get()
     {
-        using _Tp = typename ::std::remove_pointer<_Res>::type;
         __my_event.wait();
-        _Tp __local_val = __my_res[0];
-        sycl::free(__my_res, __my_exec.queue());
+        auto __local_val = __my_res[0];
```

Comment:
or may be more simple: ```return __my_res[0];```

Reply:
This is what I have just realized and changed.